client/app/widgets/Notifications/index.jsx

```diff
@@ -11,124 +11,128 @@ export type Props = {
 };
 
 export type State = {
-  notifications: string,
-  alreadyMounted: boolean,
-  open: boolean,
+  notifications?: string,
+  alreadyMounted?: boolean,
+  open?: boolean,
   modalKey?: string,
   signedInKey?: number,
 };
 
-export class Notifications extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = { notifications: '', alreadyMounted: false, open: false };
-  }
+export const Notifications = ({
+  element: propsElement,
+}: Props, {
+  notifications, alreadyMounted, open, modalKey, signedInKey,
+}: State) => {
+  // error handle : notifications, alreadyMounted, open not defined
+  let newnotifications = notifications;
+  let newalreadyMounted = alreadyMounted;
+  let newopen = open;
+  if (!newnotifications) newnotifications = '';
+  if (!newalreadyMounted) newalreadyMounted = false;
+  if (!newopen) newopen = false;
+  const [notificationsState, setnotificationsState] = useState(newnotifications);
+  const [alreadyMountedState, setalreadyMountedState] = useState(newalreadyMounted);
+  const [openState, setopenState] = useState(newopen);
+  const [signedInKeySate, setsignedInKeySate] = useState(signedInKey);
+  const [modalKeyState, setmodalKeyState] = useState(modalKey);
 
-  componentDidMount() {
-    this.fetchNotifications();
-  }
-
-  changeTitle = (count: number) => {
+  const changeTitle = (count: number) => {
     let { title } = window.document;
     const eliminate = `${title.substr(0, title.indexOf(') '))})`;
     title = title.replace(eliminate, '');
     window.document.title = count === 0 ? title : `(${count}) ${title}`;
   };
 
-  getPusherKey = (signedInKey: number) => {
-    this.setState({ signedInKey });
+  const setBody = (paramsNotifications: string[]) => {
+    let updatedNotifications = '';
+    paramsNotifications.forEach((item: string) => {
+      updatedNotifications += `<div>${item}</div>`;
+    });
+    setnotificationsState(updatedNotifications);
+  };
+
+  // eslint no-use-before-define: error
```

Comment:
Can we avoid using this and just move this variable before?

Reply:
Since setBosy function is used only once in fetchNotifications we can merge this function inside that function

Reply:
I think we should keep them separate even though it's only used once. This makes understanding the logic easier since there's a lot going on in `setBody` already.

You can just change the ordering of these functions.

Reply:
Ok.

Reply:
This should be removed.